plugins/state_history_plugin/tests/session_test.cpp

```diff
@@ -125,7 +127,20 @@ struct mock_state_history_plugin {
       return fc::time_point{};
    }
 
-   std::optional<eosio::chain::block_id_type> get_block_id(uint32_t block_num) { return block_id_for(block_num); }
+   std::optional<eosio::chain::block_id_type> get_block_id(uint32_t block_num) {
+      std::optional<eosio::chain::block_id_type> id;
+      if( trace_log ) {
+         id = trace_log->get_block_id( block_num );
+         if( id )
+            return id;
+      }
+      if( state_log ) {
+         id = state_log->get_block_id( block_num );
+         if( id )
+            return id;
+      }
```

Comment:
equivalent, but avoids repetition.

```suggestion
      for (std::optional<eosio::state_history_log>& log : { std::ref(trace_log), std::ref(state_log) }) {
         if( log ) {
            auto id = log->get_block_id( block_num );
            if( id )
               return id;
         }
      }
```

Reply:
Hmm, I think I prefer as it is. Easier to set a break-point or add a specific log statement.